pkg/splunk/enterprise/workerpool.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2018-2021 Splunk Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enterprise
+
+import (
+	"fmt"
+	"io/fs"
```

Comment:
Because of this package, the compilation is failing while trying to run UTs. In my env everything passes since my go version is 1.16.x but the CircleCI server has go version 1.13.x and I think this is why it is failing. @pdhanoyasplunk anything we can do to get around this error? Maybe update the go version on CircleCI?